### ISSUE TITLE
Error while opening CombineArchive

### ISSUE THREAD

**Issue by shalinshah1993** [SemsProject/CombineArchive#30]
Hi, I was trying to use `CombineArchive` to add omex file support in [SBSCL](https://github.com/shalinshah1993/SBSCL). However, I get error while initializing like this:

`CombineArchive archive = new CombineArchive(pathToZipFile);`.

```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at org.jdom2.input.SAXBuilder.<init>(SAXBuilder.java:338)
	at org.jdom2.input.SAXBuilder.<init>(SAXBuilder.java:221)
	at de.unirostock.sems.cbarchive.Utils.readXmlDocument(Utils.java:123)
	at de.unirostock.sems.cbarchive.CombineArchive.parseManifest(CombineArchive.java:1197)
	at de.unirostock.sems.cbarchive.CombineArchive.init(CombineArchive.java:239)
	at de.unirostock.sems.cbarchive.CombineArchive.<init>(CombineArchive.java:175)
	at org.simulator.omex.OMEXArchive.<init>(OMEXArchive.java:52)
	at org.simulator.omex.OMEXExample.main(OMEXExample.java:41)
Caused by: java.lang.UnsupportedOperationException: This parser does not support specification "null" version "null"
	at javax.xml.parsers.SAXParserFactory.setSchema(Unknown Source)
	at org.jdom2.input.sax.XMLReaders.<init>(XMLReaders.java:122)
	at org.jdom2.input.sax.XMLReaders.<clinit>(XMLReaders.java:95)
	... 8 more

```

I tried the files `12859_2014_369_MOESM1_ESM` and `12859_2014_369_MOESM2_ESM` which are available as part of this [paper](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-014-0369-z). 

It seems as if XMLParsing of the manifest file is the problem probably because the omex folder is compressed?

**Comment by binfalse** [comment]
Hi @shalinshah1993,
good news, that you want to use our library!
But bad news that you got that strange error...!?

I just tried it out myself and downloaded
* [12859_2014_369_MOESM1_ESM.zip](https://static-content.springer.com/esm/art%3A10.1186%2Fs12859-014-0369-z/MediaObjects/12859_2014_369_MOESM1_ESM.zip) from the paper,
* [CombineArchive-1.3.6-jar-with-dependencies.jar](https://bin.bio.informatik.uni-rostock.de/CombineArchive/CombineArchive-1.3.6-jar-with-dependencies.jar)

and then hacked the following tiny snippet:


```
usr@srv $ cat CaExample.java

import de.unirostock.sems.cbarchive.CombineArchive;
import java.io.File;

public class CaExample
{
    public static void main (String[] args) throws Exception
    {
        System.out.println ("test");
        CombineArchive ca = new CombineArchive (new File ("12859_2014_369_MOESM1_ESM.zip"));
        System.out.println (ca.getEntries ().size () + " entries");
    }
}
```

compiled it with 

    javac -cp CombineArchive-1.3.6-jar-with-dependencies.jar CaExample.java

and ran it:

```
usr@srv $ java -cp CombineArchive-1.3.6-jar-with-dependencies.jar:. CaExample
test
2 entries
```

So it seems to be working fine...!? I've been using OpenJDK's Java 10:

```
usr@srv $ java -version
openjdk version "10.0.1" 2018-04-17
OpenJDK Runtime Environment (build 10.0.1+10-Debian-4)
OpenJDK 64-Bit Server VM (build 10.0.1+10-Debian-4, mixed mode)
```

Can you please try and run the code above to see if that works for you?
Which Java version are you using? Which version of the CombineArchive library are you using? How do you integrate the CombineArchive library? (e.g. Maven? maybe there is a dependency problem and your tool requires something in a different version than the CombineArchive library expects it...?)
Do you have a minimal example that doesn't work for you? (which I could test/debug myself?)

**Comment by shalinshah1993** [comment]
Right now, I included `CombineArchive` in SBSCL using maven. In my pom.xml file, I have following code:

```
       <repository>
		<id>sems-maven-repository-releases</id>
		<name>SEMS Maven Repo</name>
		<url>http://mvn.sems.uni-rostock.de/releases/</url>
		<layout>default</layout>
		<snapshots>
			<enabled>false</enabled>
		</snapshots>
	</repository>

...

        <dependency>
		<groupId>de.uni-rostock.sbi</groupId>
		<artifactId>CombineArchive</artifactId>
		<version>1.3.6</version>
	</dependency>
```

May be the problem is because I need JAR with depdencies and maven jar is not standalone? 

My java version:

```
>>java --version
java 9.0.4
Java(TM) SE Runtime Environment (build 9.0.4+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.4+11, mixed mode)

>>javac --version
javac 9.0.4
```

**Comment by binfalse** [comment]
Ok, just fixed your project. The problem was JDOM2... Early JDOM2 versions are packaged as org.jdom->jdom->2.x  --- but later JDOM2 versions are packaged as org.jdom->jdom2->2.x
for maven that are two different artifacts, but classes and packages are named the same.. which obviously produces a mess...

I fixed your dependency for JDOM2 as jdom, it is now jdom2. And everything is running smoothly.

I wasn't able to implement+test it in your master, as it doesn't compile for me. but I created a branch based on 3191fc5 and also added a unit test, which compiles and shows that CAs are now working in SBSCL https://github.com/binfalse/SBSCL/tree/integrate-ca (actual commit: https://github.com/binfalse/SBSCL/commit/034b74bd58f0f842cfa13c9833f29026101cee21)

Not sure about all the other commits, but still created a pull request for you: https://github.com/shalinshah1993/SBSCL/pull/18
But maybe you just want to cherry-pick 034b74bd58f0f842cfa13c9833f29026101cee21...

does that help?
